src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/MetricsEmitter.java

```diff
@@ -847,4 +847,79 @@ public static void emitNodeMetrics(
           "Total time taken for writing {} metrics metricsdb: {}", tableName, mFinalT - mCurrT);
     }
   }
+
+
+  public static void emitShardStateMetric(
+          MetricsDB metricsDB, ShardStateMetricsSnapshot shardStateMetricsSnapshot) {
+    long mCurrT = System.currentTimeMillis();
+    Result<Record> shardStateMetrics = shardStateMetricsSnapshot.fetchAggregatedShardStateMetrics();
+    List<String> dims =
+            new ArrayList<String>() {
+              {
+                this.add(AllMetrics.ShardStateDimension.INDEX_NAME.toString());
+                this.add(AllMetrics.ShardStateDimension.SHARD_ID.toString());
+                this.add(AllMetrics.ShardStateDimension.SHARD_TYPE.toString());
+                this.add(AllMetrics.ShardStateDimension.NODE_NAME.toString());
+                this.add(AllMetrics.ShardStateDimension.SHARD_STATE.toString());
+              }
+            };
```

Comment:
`dims` can be made a static variable rather than initializing an array list on each `emitShardStateMetric` execution. 

Refer [this](https://github.com/opendistro-for-elasticsearch/performance-analyzer-rca/blob/d59004072522d90f307aebf1bf0b4df4619b5744/src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/MetricsEmitter.java#L77)

Reply:
done